lib/dolly/document.rb

```diff
@@ -98,6 +98,10 @@ def id_as_resource
       CGI::escape id
     end
 
+    def attach_file file_name, mime_type, body
+      database.attach id_as_resource, file_name, body, { 'Content-Type' => mime_type }
```

Comment:
I think having the user supply the mime type makes the most sense. if you feel differently, please discuss.